src/App.tsx

```diff
@@ -1,27 +1,100 @@
-import React from 'react';
+import classnames from 'classnames';
+import React, { useState } from 'react';
+import { get5First, getAll, getRedGoods } from './api/goods';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+enum Tab {
+  None = 'none',
+  All = 'all',
+  First5 = 'first5',
+  Red = 'Red',
+}
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [currentTab, setCurrentTab] = useState(Tab.None);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handleGoodsLoad = (
+    getGoods: () => Promise<Good[]>,
+    tab: Tab,
+  ) => {
+    return () => {
+      if (tab !== currentTab) {
+        setIsLoading(true);
+        setIsError(false);
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+        getGoods()
+          .then(setGoods)
+          .catch(() => setIsError(true))
+          .finally(() => setIsLoading(false));
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+        setCurrentTab(tab);
+      }
+    };
+  };
 
-    <GoodsList goods={[]} />
-  </div>
-);
+  return (
+    <div className="App">
+
+      <div className="panel is-primary is-flex is-flex-direction-column">
+        <h1
+          className="panel-heading has-text-centered"
+        >
+          Dynamic list of Goods
+        </h1>
+
+        <p className="panel-tabs p-2 has-background-primary">
+          <button
+            type="button"
+            className={classnames(
+              'button',
+              'is-light',
+              'is-outlined',
+              'mr-5',
+              { 'is-loading': isLoading && currentTab === Tab.All },
+            )}
+            data-cy="all-button"
+            onClick={handleGoodsLoad(getAll, Tab.All)}
```

Comment:
```suggestion
            onClick={handleGoodsLoad(getAll)}
```
You can omit the second argument - when the button is disabled user can't trigger action.